Lib/_pyrepl/_module_completer.py

```diff
@@ -98,7 +100,14 @@ def _find_modules(self, path: str, prefix: str) -> list[str]:
                        if mod_info.ispkg and mod_info.name == segment]
             modules = self.iter_submodules(modules)
         return [module.name for module in modules
-                if module.name.startswith(prefix)]
+                if self.is_suggestion_match(module.name, prefix)]
+
+    def is_suggestion_match(self, module_name: str, prefix: str) -> bool:
+        if prefix:
+            return module_name.startswith(prefix)
+        # Return False if no prefix and the module is private
+        # So that the default behavior is only returning public modules
```

Comment:
Lukasz mentioned we shouldn't be commenting *what* the code is doing but rather *why* it's doing what it's doing. I think this makes it clearer:

```suggestion
        # For consistency with attribute completion, which
        # does not suggest private attributes unless requested.
```